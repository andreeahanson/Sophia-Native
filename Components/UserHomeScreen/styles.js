import { StyleSheet } from 'react-native';
import theme from '../../theme';

export default StyleSheet.create({
	greeting: {
		fontSize: 30
	}
});
