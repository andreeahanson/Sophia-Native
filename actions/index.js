export const loadProfile = profile => ({
  type: "LOAD_PROFILE",
  profile
})

export const loadLists = lists => ({
  type: "LOAD_LISTS",
  lists
})

export const addList = newList => ({
  type: "ADD_LIST",
  newList
})

export const deleteList = listId => ({
  type: "DELETE_LIST",
  listId
})

export const loadTasks = (listId, tasks) => ({
  type: "LOAD_TASKS",
  listId,
  tasks
})

export const addTask = newTask => ({
  type: "ADD_TASK",
  newTask
})

export const deleteTask = taskId => ({
  type: "DELETE_TASK",
  taskId
})

export const editTask = (nameToChange, taskId) => ({
  type: "EDIT_TASK",
  nameToChange,
  taskId
})