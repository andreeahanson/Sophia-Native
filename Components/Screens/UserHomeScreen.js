import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { TouchableHighlight } from 'react-native-gesture-handler';
import { connect } from 'react-redux';
import { loadLists } from '../../actions';
import { PropTypes } from 'prop-types';
import { styles } from './styleUserHomeScreen';

export class UserHomeScreen extends Component {
	render () {
		return (
			<View style={styles.container}>
				<View style={styles.headerContainer}>
					<Text style={styles.header} accessibilityLabel="Speech Operated Personal Household Interactive Assistant">
						SOPHIA
					</Text>
				</View>
				<Text style={styles.greeting}>Welcome Back, {this.props.user.name}!</Text>
				<View style={styles.routes}>
					<TouchableHighlight
						underlayColor="black"
						accessibilityLabel="Tap me to navigate to your profile. From there, view your personal information"
						accessible={true}
						onPress={() => this.props.navigation.navigate('Profile', this.props.user)}
						style={styles.touchExpander}>
						<Text style={styles.button}>My Account</Text>
					</TouchableHighlight>
				</View>
				<View style={styles.routes}>
					<TouchableHighlight
						underlayColor="black"
						accessibilityLabel="Tap me to navigate to your todo lists. From there view or create your tasks."
						accessible={true}
						onPress={() => this.props.navigation.navigate('Lists', this.props.lists)}
						style={styles.touchExpander}>
						<Text style={styles.button}>My Lists</Text>
					</TouchableHighlight>
				</View>
			</View>
		);
	}
}

export const mapStateToProps = state => ({
	user: state.userAccount,
	lists: state.lists
});

export const mapDispatchToProps = dispatch => ({
	loadLists: lists => dispatch(loadLists(lists))
});

export default connect(mapStateToProps, mapDispatchToProps)(UserHomeScreen);

UserHomeScreen.propTypes = {
	userAccount: PropTypes.object,
	lists: PropTypes.array
};
