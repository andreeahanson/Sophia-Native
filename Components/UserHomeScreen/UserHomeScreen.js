import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Header } from 'react-native-elements';
import { connect } from 'react-redux';
import { loadLists } from '../../actions';
import { PropTypes } from 'prop-types';
import styles from './styles';
import Button from '../common/Button/Button';
import { logOut } from '../../actions';

export class UserHomeScreen extends Component {

	static navigationOptions = ({ navigation }) => {
		const { params = {} } = navigation.state;
		const headerRight = (
			<Button accessibilityLabel="Tap to log out" onPress={params.logOut} style={{ borderRadius: 5, height: 20, width: '20%' }}>
			Log Out
		</Button>
		);

		return headerRight;
	};

	componentDidMount = () => {
		this.props.navigation.setParams({logout: this.logOut})
	}
	logOut = () => {
		this.props.logOut();
		this.props.navigation.navigate('Home');
	};

	render() {
		const { user, navigation } = this.props;
		return (
			<View style={{...styles.container}}>
				<Header accessibilityLabel="Speech Operated Personal Household Interactive Assistant">SOPHIA</Header>
				<Text style={styles.greeting}>Welcome Back, {'\n' + user.name}!</Text>
				<Button
					accessibilityLabel="Tap me to navigate to your profile. From there, view your personal information"
					onPress={() => navigation.navigate('Profile', user)}
				>
					My Profile
				</Button>
				<Button
					accessibilityLabel="Tap me to navigate to your todo lists. From there view or create your tasks."
					onPress={() => {
						user.role === "client" ? navigation.navigate('NeedDone') : navigation.navigate('CaretakerList');
					}}
				>
					Things I Need Done
				</Button>
				<Button
					accessibilityLabel="Tap me to navigate to your todo lists. From there view or create your tasks."
					onPress={() => {
						navigation.navigate('NeedToDo');
					}}
				>
					Things I Need To Do
				</Button>
			</View>
		);
	}
}
export const mapStateToProps = state => ({
	user: state.userAccount,
	lists: state.lists,
});
export const mapDispatchToProps = dispatch => ({
	loadLists: lists => dispatch(loadLists(lists)),
	logOut: () => dispatch(logOut())
});
export default connect(
	mapStateToProps,
	mapDispatchToProps
)(UserHomeScreen);

UserHomeScreen.propTypes = {
	userAccount: PropTypes.object,
	lists: PropTypes.array,
};
