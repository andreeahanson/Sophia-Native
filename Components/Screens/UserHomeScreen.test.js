import React from "react";
import { shallow } from "enzyme";
import { UserHomeScreen } from "./UserHomeScreen";
import "react-native";
import { mapStateToProps, mapDispatchToProps } from "./UserHomeScreen";
import { loadLists } from "../../actions";

jest.mock("react-native-gesture-handler", () => {
  return {}
});

let wrapper = shallow(<UserHomeScreen 
user = { {name: "Test Name"} }
/>)

test("UserHomeScreen renders correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

it("should return a userAccount object", () => {
  const initialState = {}

  let mappedProps = mapStateToProps(initialState);

  expect(mappedProps).toEqual(initialState);
});

it("calls dispatch with an array of lists when loadList action is called", () => {
  let initialState = {
    lists: []
  };

  const mockDispatch = jest.fn();
  const mockAction = loadLists(initialState.lists);

  const mappedProps = mapDispatchToProps(mockDispatch);
  mappedProps.loadLists(initialState.lists);

  expect(mockDispatch).toHaveBeenCalledWith(mockAction);
});