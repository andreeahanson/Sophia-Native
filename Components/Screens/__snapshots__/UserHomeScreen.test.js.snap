// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHomeScreen renders correctly 1`] = `ShallowWrapper {}`;
